src/Sentry/Internal/Http/HttpTransport.cs

```diff
@@ -31,14 +37,62 @@ internal class HttpTransport : ITransport
 
         public async Task SendEnvelopeAsync(Envelope envelope, CancellationToken cancellationToken = default)
         {
-            var request = CreateRequest(envelope);
+            var instant = DateTimeOffset.Now;
+
+            // Discard envelope items if they don't fit in the quota limit
+            var envelopeItems = new List<EnvelopeItem>();
+            foreach (var envelopeItem in envelope.Items)
+            {
+                var category = envelopeItem.TryGetType();
+
+                if (!string.IsNullOrWhiteSpace(category) &&
+                    _quotaLimitResets.TryGetValue(category, out var resetInstant))
+                {
+                    if (instant > resetInstant)
+                    {
+                        envelopeItems.Add(envelopeItem);
+                    }
```

Comment:
Could we have some `LogDebug` to see what's going on in debug mode please?

Reply:
LogDebug or also LogInfo?